src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -140,39 +117,32 @@ public static Logger getLogger() {
 
   public static void gameOverClearCache() {
     // Is static, set to null so that we don't keep the data around after a game is exited
-    s_battleCalculator.setGameData(null);
+    ProBattleUtils.setData(null);
     ProLogUI.clearCachedInstances();
   }
 
-  public IOddsCalculator getCalc() {
-    return s_battleCalculator;
-  }
-
-  @Override
-  public final GameData getGameData() {
-    if (data != null) {
-      return data;
-    } else {
-      return super.getGameData();
-    }
-  }
-
   @Override
   public void stopGame() {
     super.stopGame(); // absolutely MUST call super.stopGame() first
-    s_battleCalculator.cancel(); // cancel any current calcing
+    ProBattleUtils.cancel(); // cancel any current calcing
   }
 
   public void setStoredStrafingTerritories(final List<Territory> strafingTerritories) {
     storedStrafingTerritories = strafingTerritories;
   }
 
+  private void initializeData() {
```

Comment:
High level constants , yes as static. For `getGameData()` and `this`, no. IOC is paraphrased as the "don't call us, we'll call you" principle. Meaning, this class should not set any data on ProData, or for example ProBattleUtils, but instead should just consume those objects.

The example on line 144 and 145 is perhaps a bit more clear. IOC would basically have the "ProBattleUtils" object passed as a parameter to whatever needs it. Whatever is calling the `move` mehod here would be responsible for the call to `intiializeData()` on line 144 and the next call on line 145, and *then* it would call `move` method. The difference there is the `move` method has no idea about the static data state of ProBattleUtils, and it is more focused on performing the business logic.